idaes/surrogate/pysmo/sampling.py

```diff
@@ -720,29 +819,23 @@ def __init__(self, data_input, number_of_samples=None, sampling_type=None):
         print('Sampling type: ', self.sampling_type, '\n')
 
         if self.sampling_type == 'selection':
-            if isinstance(data_input, pd.DataFrame):
-                data = data_input.values
-                data_headers = data_input.columns.values.tolist()
-            elif isinstance(data_input, np.ndarray):
-                data = data_input
-                data_headers = []
+            if isinstance(data_input, (pd.DataFrame, np.ndarray)):
+                self.selection_columns_preprocessing(data_input, xlabels, ylabels)
             else:
-                raise ValueError('Pandas dataframe or numpy array required.')
-            self.data = data
-            self.data_headers = data_headers
+                raise ValueError('Pandas dataframe or numpy array required for sampling_type "selection."')
 
             # Catch potential errors in number_of_samples
             if number_of_samples is None:
                 print("\nNo entry for number of samples to be generated. The default value of 5 will be used.")
```

Comment:
General question/observation: Should we be using the IDAES logger for this?

Reply:
At the time this script was written we were unsure whether PySMO would be part of IDAES or Pyomo so I avoided the IDAES logger. Thus, the script is has a number of print statements and I thought I'd keep it uniform for now.

The plan is that all those print statements will sometime in the future be replaced with the IDAES logger in one go (if I ever get the time).